src/modes/direct_select.js

```diff
@@ -127,6 +127,27 @@ DirectSelect.clickActiveFeature = function (state) {
   state.feature.changed();
 };
 
+
+DirectSelect.onDblClick = function(state, e) {
+  const { feature, selectedCoordPaths } = state;
+
+  if (!e.featureTarget
+    || feature.type !== 'LineString'
+    || selectedCoordPaths.length !== 1
+    || (Number(selectedCoordPaths[0]) !== 0
+      && Number(selectedCoordPaths[0]) !== feature.coordinates.length - 1)
```

Comment:
**suggestion**

Assign these conditionals to variable so we know what they mean